api/v1alpha1/sparkcluster_webhook.go

```diff
@@ -110,57 +159,193 @@ func (r *SparkCluster) Default() {
 
-	return r.validateSparkCluster()
+	return sc.validateSparkCluster()
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
-func (r *SparkCluster) ValidateUpdate(old runtime.Object) error {
-	sparkLogger.WithValues("sparkcluster", client.ObjectKeyFromObject(r)).Info("validating update")
+func (sc *SparkCluster) ValidateUpdate(old runtime.Object) error {
+	sparkLogger.WithValues("sparkcluster", client.ObjectKeyFromObject(sc)).Info("validating update")
 
-	return r.validateSparkCluster()
+	return sc.validateSparkCluster()
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
 // Not used, just here for interface compliance.
-func (r *SparkCluster) ValidateDelete() error {
+func (sc *SparkCluster) ValidateDelete() error {
 	return nil
 }
 
-func (r *SparkCluster) validateSparkCluster() error {
+func (sc *SparkCluster) validateSparkCluster() error {
 	var allErrs field.ErrorList
 
-	if err := r.validateWorkerReplicas(); err != nil {
+	if err := sc.validateMutualTLSMode(); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := sc.validateWorkerReplicas(); err != nil {
 		allErrs = append(allErrs, err)
 	}
-	if err := r.validateWorkerResourceRequestsCPU(); err != nil {
+	if err := sc.validateWorkerMemoryLimit(); err != nil {
 		allErrs = append(allErrs, err)
 	}
-	if errs := r.validatePorts(); errs != nil {
+	if err := sc.validateWorkerResourceRequestsCPU(); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if errs := sc.validatePorts(); errs != nil {
+		allErrs = append(allErrs, errs...)
+	}
+	if errs := sc.validateDriverConfigs(); errs != nil {
+		allErrs = append(allErrs, errs...)
+	}
+	if errs := sc.validateAutoscaler(); errs != nil {
+		allErrs = append(allErrs, errs...)
+	}
+	if errs := sc.validateImage(); errs != nil {
 		allErrs = append(allErrs, errs...)
 	}
-	if errs := r.validateAutoscaler(); errs != nil {
+	if errs := sc.validateFrameworkConfigs(); errs != nil {
 		allErrs = append(allErrs, errs...)
 	}
-	if errs := r.validateImage(); errs != nil {
+	if errs := sc.validateKeyTabConfigs(); errs != nil {
 		allErrs = append(allErrs, errs...)
 	}
+	if err := sc.validateNetworkPolicies(); err != nil {
+		allErrs = append(allErrs, err)
+	}
 
 	if len(allErrs) == 0 {
 		return nil
 	}
 
 	return apierrors.NewInvalid(
 		schema.GroupKind{Group: "distributed-compute.dominodatalab.com", Kind: "SparkCluster"},
-		r.Name,
+		sc.Name,
 		allErrs,
 	)
 }
 
-func (r *SparkCluster) validateWorkerReplicas() *field.Error {
-	replicas := r.Spec.Worker.Replicas
+func (sc *SparkCluster) validateNetworkPolicies() *field.Error {
+	if sc.Spec.NetworkPolicy.ExternalPolicyEnabled != nil &&
+		*sc.Spec.NetworkPolicy.ExternalPolicyEnabled &&
+		len(sc.Spec.NetworkPolicy.ExternalPodLabels) == 0 {
+		return field.Invalid(
+			field.NewPath("spec").Child("NetworkPolicy").Child("ExternalPodLabels"),
```

Comment:
```suggestion
			field.NewPath("spec").Child("networkPolicy").Child("externalPodLabels"),
```

I believe these should paths should be _camelCase_. Something else I found while working on Dask, `field.NewPath` accepts "more names".

e.g.

```go
field.NewPath("spec", "networkPolicy", "externalPodLabels")
```